### DISCUSSION TITLE
upgrade - error python migrate

### DISCUSSION THREAD

**Discussion by mrhhats** [mediacms-io/mediacms#736] [Q&A]
Hey guys, running 1.6 on ubuntu. Trying the upgrade instructions and getting this error. Thank you in advance for any assistance!

upgrade instructions following:

Update
If you've used the above way to install MediaCMS, update with the following:

cd /home/mediacms.io/mediacms # enter mediacms directory
source  /home/mediacms.io/bin/activate # use virtualenv
git pull # update code
python manage.py migrate # run Django migrations
sudo systemctl restart mediacms celery_long celery_short # restart services

**--terminal output--

sudo python manage.py migrate
[sudo] password for jason: 
  File "manage.py", line 12
    raise ImportError("Couldn't import Django. Are you sure it's installed and " "available on your PYTHONPATH environment variable? Did you " "forget to activate a virtual environment?") from exc
                                                                                                                                                                                               ^
SyntaxError: invalid syntax**

**Comment by mgogoulos** [comment]
Hi, these instructions assume you have installed MediaCMS as root and you are now running the commands as root.
The fact that you use sudo makes me think that while you are logged in as user and enable Python's virtualenv (via `source /home/mediacms.io/bin/activate`), when you run the next command via sudo (`sudo python manage.py migrate`) you are outside of the virtualenv, and Python compains. 

Try out to `sudo su` and then run all commands

**Comment by mrhhats** [comment]
Thank you sir, that was the trick! all commands ran without error. One quick question. I'm seeing a version.py file and after all the commands its still showing 1.6 inside that file.

**Reply by mgogoulos** [reply]
Glad that this worked!

Don't bother about versions.py, it's not getting updated (the whole naming and how it works needs to be reconsidered)

**Comment by MonaWisers3c** [comment]
It looks like there is a syntax error in the "manage.py" file, which is causing the "python manage.py migrate" command to fail during the upgrade process of MediaCMS on Ubuntu 1.6.

**Comment by FredrickZywiecG** [comment]
It looks like the error you're encountering is a syntax error in the manage.py file, which suggests there may be an issue with the code itself rather than the upgrade instructions. You might want to check the file and make sure it's not missing any necessary dependencies or has any other errors in the code. Additionally, double-check that Django is installed and available on your PYTHONPATH environment variable, and that your virtual environment is activated.

**Comment by mgogoulos** [comment]
@MonaWisers3c @joannwillia34 @FredrickZywiecG @ElianaDemaray0B are you powered by ChatGPT, or is it another LLM?
